.github/workflows/rust.yml

```diff
@@ -92,27 +82,53 @@ jobs:
           --health-timeout 5s
           --health-retries 5
     steps:
-      - name:                   Checkout repository
-        uses:                   actions/checkout@v2
-      - uses: actions/cache@v3
+      - uses:                   actions/checkout@v2
+      - run: rustup toolchain install stable --profile minimal
+      - uses: Swatinem/rust-cache@v2
         with:
-          path: |
-            ~/.cargo/bin/
-            ~/.cargo/registry/index/
-            ~/.cargo/registry/cache/
-            ~/.cargo/git/db/
-            target/
-          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
+          key: "coverage"
       - name: Setup database
         run: |
           cargo install sqlx-cli --no-default-features --features native-tls,postgres
           cargo sqlx database setup
       - name:                   Generate code coverage
         run: |
-          cargo +nightly tarpaulin --verbose --all-features --workspace --timeout 120 --out Xml
+          cargo tarpaulin --all-targets --skip-clean --workspace --timeout 120 --out Xml
       - name:                   Upload to codecov.io
         uses:                   codecov/codecov-action@v2
         with:
           token:                ${{secrets.CODECOV_TOKEN}}
           fail_ci_if_error:     true
 
+
+  test:
+    runs-on: ubuntu-latest
+    container: rust:latest
+    defaults:
+      run:
+        working-directory: ./backend
+    services:
+      postgres:
+        image: postgres
+        env:
+          POSTGRES_PASSWORD: password
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+
+    steps:
+    - uses: actions/checkout@v3
+    - run: rustup toolchain install stable --profile minimal
+    - uses: Swatinem/rust-cache@v2
+      with:
+        key: "test"
+    - name: Setup database
+      run: |
+        cargo install sqlx-cli --no-default-features --features native-tls,postgres
+        cargo sqlx database setup
+    - name: Check docs
```

Comment:
this more a compilation thing than a test (and thus would e more appropriate in the other task)

Reply:
Oh, ok. I thought that the command sets up the database and is needed for the following commands to work

Reply:
I am takling about the "check docs"